src/MOCKernel.cpp

```diff
@@ -0,0 +1,165 @@
+#include "MOCKernel.h"
+#include "TrackGenerator.h"
+
+/**
+ * @brief Constructor for the MOCKernel assigns default values
+ * @param track_generator the TrackGenerator used to pull relevant tracking
+ *        data from
+ * @param row_num the row index into the temporary segments matrix
+ */
+MOCKernel::MOCKernel(TrackGenerator* track_generator, int row_num) {
+  _count = 0;
+  _max_tau = track_generator->retrieveMaxOpticalLength();
+}
+
+
+/**
+ * @brief Constructor for the VolumeKernel assigns default values, calls
+ *        the MOCKernel constructor, and pulls refernces to FSR locks and FSR
+ *        volumes from the provided TrackGenerator.
+ * @param track_generator the TrackGenerator used to pull relevant tracking
+ *        data from
+ * @param row_num the row index into the temporary segments matrix
+ */
+VolumeKernel::VolumeKernel(TrackGenerator* track_generator, int row_num) :
+                           MOCKernel(track_generator, row_num) {
+  _FSR_locks = track_generator->getFSRLocks();
+
+  if (_FSR_locks == NULL)
```

Comment:
Add that as a comment then. And perhaps reword the error message to reflect this? Otherwise, no user will understand 1) what the issue is or 2) how to fix it if they see this issue, even though it is as simple as generating tracks before setting this up.